query-engine/connectors/connector/src/new_interfaces/mod.rs

```diff
@@ -0,0 +1,24 @@
+use prisma_models::*;
+
+pub trait Connector {
+    fn with_transaction<F, T>(&self, f: F) -> crate::Result<T>
+    where
+        F: FnOnce(&mut dyn MaybeTransaction) -> crate::Result<T>;
+
+    //    fn with_connection<F, T>(&self, f: F) -> crate::Result<T>
+    //    where
+    //        F: FnOnce(&mut dyn MaybeTransaction) -> crate::Result<T>;
+}
+
+pub trait MaybeTransaction: ReadOperations + WriteOperations {}
```

Comment:
The reason can be seen above in the commented code. In the long run i think we will expose `with_transaction` and `with_connection`. The APIs of both are absolutely identical right now. I wanted to indicate with the name that it might be either a Transaction or a Connection.
Not sure whether this is the best way though.

Reply:
No, it's a fine abstraction for now.